test/unit/RelayServer.test.ts

```diff
@@ -1,688 +1,535 @@
+  const fakeEstimationBeforeFees = 100000;
+  const TOKEN_X_RATE = '0.5';
+  const GAS_PRICE = 10000;
+  const FAKE_GAS_FEE_PERCENTAGE = 0.1;
+  const FAKE_TRANSFER_FEE_PERCENTAGE = 0.1;
+  const TOKEN_AMOUNT_TO_TRANSFER = '1000000000000000000'; //18 zeros
+  const TOKEN_AMOUNT_IN_REQUEST = '500000000000000000'; //17 zeros
+
+  let relayServer: RelayServer;
+  let mockServer: sinon.SinonMock;
+  let dataWhenTransfer: string;
+  let dataWhenTransferFrom: string;
+  let expectedFeeFromTransfer: BigNumberJs;
+
+  beforeEach(function () {
+    //Build and mock server
+    const managerKeyManager = createStubInstance(KeyManager);
+    const workersKeyManager = createStubInstance(KeyManager);
+    const txStoreManager = createStubInstance(TxStoreManager);
+
+    relayServer = new RelayServer({
+      managerKeyManager,
+      txStoreManager,
+      workersKeyManager,
     });
 
-    afterEach(() => {
-        restore();
+    mockServer = mock(relayServer);
+
+    //Set stubs
+    sinon.replaceGetter(rifClient, 'estimateRelayMaxPossibleGas', () =>
+      sinon.stub().resolves(BigNumber.from(fakeEstimationBeforeFees))
+    );
+    sinon.replaceGetter(rifClient, 'standardMaxPossibleGasEstimation', () =>
+      sinon.stub().resolves(BigNumber.from(fakeEstimationBeforeFees))
+    );
+
+    sinon.stub(utils, 'getProvider').returns(providers.getDefaultProvider());
+
+    const token = {
+      name: () => Promise.resolve('TestToken'),
+      symbol: () => Promise.resolve('TT'),
+      decimals: () => Promise.resolve(18),
+    } as unknown as ERC20;
+    sinon.stub(ERC20__factory, 'connect').returns(token);
+
+    sinon.stub(Conversions, 'getXRateFor').resolves(TOKEN_X_RATE);
+
+    //Build the data for the transfer() and transferFrom()
+    const fakeFromAddress =
+      '000000000000000000000000e87286ba960fa7aaa5b376083a31d440c8cb4bc8';
+    const fakeToAddress =
+      '0000000000000000000000008470af7f41ee2788eaa4cfc251927877b659cdc5';
+    const tokenAmountToTransferAsHex = BigNumber.from(TOKEN_AMOUNT_TO_TRANSFER)
+      .toHexString()
+      .substring(2) //removes 0x
+      .padStart(64, '0'); //fills with 0 to the left
+
+    dataWhenTransfer =
+      '0x' + TRANSFER_HASH + fakeToAddress + tokenAmountToTransferAsHex;
+
+    dataWhenTransferFrom =
+      '0x' +
+      TRANSFER_FROM_HASH +
+      fakeFromAddress +
+      fakeToAddress +
+      tokenAmountToTransferAsHex;
+
+    //Calculate the expected fee value when a transfer/transferFrom is executed
+    const tokenFee = BigNumberJs(TOKEN_AMOUNT_TO_TRANSFER).multipliedBy(
+      FAKE_TRANSFER_FEE_PERCENTAGE
+    );
+    const tokenFeeAsFraction = toPrecision({
+      value: tokenFee,
+      precision: -18,
     });
+    const feeAsFractionInNative = tokenFeeAsFraction.multipliedBy(TOKEN_X_RATE);
+    const feeInNative = toPrecision({
+      value: feeAsFractionInNative,
+      precision: 18,
+    });
+    expectedFeeFromTransfer = feeInNative.dividedBy(GAS_PRICE);
```

Comment:
I moved some code to a before section. Perhaps the code related to stubs and mocks is better to keep it here since there are values setted up specifically for some tests and after each one, I'm running sinon.restore().

Reply:
Yes, I totally agree 👍🏽